### PR TITLE
Process/withdrawal

### PR DESCRIPTION
**Pull request by IgboPharaoh**
This pr introduces the structure of the withdrawal process and steps taken to make a successful withdrawal:  
-Onclick of the withdraw button a modal pops up
-The modal ascertains users are sure they want to perform an action. This modal show a brief summary of the transaction they want to perform.  
-An API call is made to start the withdrawal process. 
-A toast component is shown which notifies the users that the withdrawal transaction has started and directs them to view the transaction in the transaction history section. 
-The transaction history section introduces the TransactionInfo component which shows a summary or overview of the transaction made.
-Concurrently the status of the transaction is being watched at intervals and updated as confirmations reach the required number of confirmations

### CODE REVIEW

File: src/components/WithdrawTab.tsx [hunk 1]
```diff
@@ -1,29 +1,73 @@
...
 	amount: number;
-	walletAddress: string;
+	address: string;
 }
 
 export interface WithdrawTabProps {
 	federationId: string;
 }
 
+const truncateStringFormat = (arg: string): string => {
```
Comment (okjodom) [inline src/components/WithdrawTab.tsx:51]: DRY: Can we [share the same helper util from deposit tab](https://github.com/GETLN/mintgate/blob/main/src/components/DepositTab.tsx#L25)?

File: src/components/WithdrawTab.tsx [hunk 2]
```diff
@@ -35,16 +79,41 @@ export const WithdrawTab = React.memo(function WithdrawTab({
...
 			const txId = await mintgate.requestWithdrawal(
 				federationId,
 				amount,
-				walletAddress
+				address
 			);
-			console.log(txId);
+
+			console.log(txId, 'trasaction id:::');
+			setTransactionList([
```
Comment (okjodom) [inline src/components/WithdrawTab.tsx:109]: Should we show the newest transaction at the top of the list?

File: src/components/WithdrawTab.tsx [hunk 3]
```diff
@@ -81,6 +150,202 @@ export const WithdrawTab = React.memo(function WithdrawTab({
...
+									p={{ base: '10px', md: '13px', lg: '16px' }}
+								>
+									Proceed
+								</Button>
+							</ModalFooter>
+						</ModalContent>
+					</Modal>
+				)}
+
+				{step === 1 && (
```
Comment (okjodom) [inline src/components/WithdrawTab.tsx]: Are we still showing multiple modals within the withdraw flow?

**Review by okjodom — APPROVED** [review]
Left some nit comments about code improvements.
For some of the comments seem fixed with the last commit, but were relevant in earlier commits. It's okay to resolve these ones, though I would have expected a git amend of earlier  commits, rather than adding later commits to correct changes introduced in the preceding ones. It's confusing :)

**Comment by okjodom** [comment]
Approved, and merging. We can follow up with polish and logic changes

File: src/components/WithdrawTab.tsx (hunk 3 above)
Comment (IgboPharaoh) [reply src/components/WithdrawTab.tsx]: this was actually removed even from my end. Probably you may be viewing the penultimate commit

**Pull request merged**